test/Microsoft.ML.Tests/OnnxSequenceTypeWithAttributesTest.cs

```diff
@@ -63,7 +63,45 @@ public void OnnxSequenceTypeWithColumnNameAttributeTest()
             {
                 Assert.Equal(onnxOut[keys[i]], input.Input[i]);
             }
+        }
+
+        public class WrongOutputObj
+        {
+            [ColumnName("output")]
+            [OnnxSequenceType(typeof(IEnumerable<float>))]
+            public IEnumerable<float> Output;
+        }
+
+        public static PredictionEngine<FloatInput, WrongOutputObj> LoadModelWithWrongCustomType(string onnxModelFilePath)
+        {
+            var ctx = new MLContext(1);
+            var dataView = ctx.Data.LoadFromEnumerable(new List<FloatInput>());
 
+            var pipeline = ctx.Transforms.ApplyOnnxModel(
+                                modelFile: onnxModelFilePath,
+                                outputColumnNames: new[] { "output" }, inputColumnNames: new[] { "input" });
+
+            var model = pipeline.Fit(dataView);
+            return ctx.Model.CreatePredictionEngine<FloatInput, WrongOutputObj>(model);
+        }
+
+        [OnnxFact]
+        public void OnnxSequenceTypeWithColumnNameAttributeTestWithWrongCustomType()
+        {
+            var modelFile = Path.Combine(Directory.GetCurrentDirectory(), "zipmap", "TestZipMapString.onnx");
+            //var predictor = LoadModelWithWrongCustomType(modelFile);
+            var expectedExceptionMessage = "The expected type 'System.Collections.Generic.IEnumerable`1[System.Collections.Generic.IDictionary`2[System.String,System.Single]]'" +
+                " does not match the type of the 'output' member: 'System.Collections.Generic.IEnumerable`1[System.Single]'." +
+                " Please change the output member to 'System.Collections.Generic.IEnumerable`1[System.Collections.Generic.IDictionary`2[System.String,System.Single]]'\r\nParameter name: actualType";
+            try
+            {
+                var predictor = LoadModelWithWrongCustomType(modelFile);
+            }
+            catch (System.Exception ex)
+            {
+                Assert.Equal(ex.Message, expectedExceptionMessage);
```

Comment:
This is in reverse order. It should be `Assert.Equal(expectedExceptionMessage, ex.Message)` (i.e. the expected value should come first when using Assert.Equal).

BTW, the CI is failing because it seems you'll need to remove the carriage return and newline char (`\r\n`) from your expected error message:

```
2020-05-28T00:31:53.7109793Z   X Microsoft.ML.Tests.OnnxSequenceTypeWithAttributesTest.OnnxSequenceTypeWithColumnNameAttributeTestWithWrongCustomType [45ms]
2020-05-28T00:31:53.7110430Z   Error Message:
2020-05-28T00:31:53.7110857Z    Assert.Equal() Failure
2020-05-28T00:31:53.7122538Z                                  â†“ (pos 387)
2020-05-28T00:31:53.7123614Z Expected: Â·Â·Â·ing,System.Single]]' (Parameter 'actualType')
2020-05-28T00:31:53.7124081Z Actual:   Â·Â·Â·ing,System.Single]]'\r\nParameter name: actualType
2020-05-28T00:31:53.7124490Z                                  â†‘ (pos 387)
2020-05-28T00:31:53.7124782Z   Stack Trace:
2020-05-28T00:31:53.7125451Z      at Microsoft.ML.Tests.OnnxSequenceTypeWithAttributesTest.OnnxSequenceTypeWithColumnNameAttributeTestWithWrongCustomType() in D:\a\1\s\test\Microsoft.ML.Tests\OnnxSequenceTypeWithAttributesTest.cs:line 102
```

Reply:
That works locally and some of the pipelines(Windows). I guess the reason could be that windows and linux have different way to encode wstring. Let me truncate the string before comparing it as this is not critical in this case.